### PR TITLE
Added README for author synonyms

### PR DESCRIPTION
**Pull request by ivssh**
- [x] Add introduction of analyzer in Elasticsearch
- [x] Add what synonyms filter is
- [x] Add synonyms logic
- [x] Difference between index-time and search-time synonyms
- [x] Expansion of synonyms

### CODE REVIEW

**Comment by ivssh** [comment]
@adamjonas Please review the README and let me know if anything that's unclear or needs to be amended.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        "filter": {
+          "synonym_filter": {
+            "type": "synonym",
+            "synonyms": [
+              "PS => PlayStation",
+              "Play Station => PlayStation"
+            ]
+          }
+
+For this example, explicit mappings are used which means the token on the lefthand side of => is replaced with the one on the right side. We will use equivalent synonyms later, which means the tokens provided are treated equivalently.
```
Comment (nassersaazi) [inline README.md]: ``
```suggestion
In this example, explicit mappings are used which means the token on the left side of => is replaced with the one on the right. We will use equivalent synonyms later, which means the tokens provided are treated equivalently.
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+The synonym filter, it’s a bit special and may be surprising to many of us. In this example, even though the synonym_filter filter is put after the lowercase filter, the tokens returned by this filter are also passed to the lowercase filter and thus also get lowercased. Therefore, you don’t need to provide lowercase tokens in the synonym list or in the synonym file.
```
Comment (nassersaazi) [inline README.md]:
```suggestion
The synonym filter, is a bit special and may be surprising to many of us. In this example, even though the synonym_filter filter is put after the lowercase filter, the tokens returned by this filter are also passed to the lowercase filter and thus also get lowercased. Therefore, you don’t need to provide lowercase tokens in the synonym list or in the synonym file.
```

**Review by nassersaazi — COMMENTED** [review]
LGTM. just some nits regarding grammar in the description

File: README.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+We can use the `_analyze` endpoint to debug what tokens are emitted by each analyzer.
+```sh
+GET bitcoin-search-index-revamped/_analyze
+{
+  "text": "Gregory Maxwell",
+  "analyzer": "analyzer_search",
+  "explain": true
+}
+```
+The output of the above in `Dev Tools` is as follows:
```
Comment (nassersaazi) [inline README.md]:
```suggestion
The output of the above query in `Dev Tools` is as follows:
```

**Pull request merged**